upload_components/upload.sh

```diff
@@ -17,17 +33,25 @@ for ITEM in "${DIRECTORIES[@]}"; do
         NAME=$(basename "$(realpath "${FULL_PATH}")")
     fi
 
+    echo "Creating namespace ${NAMESPACE}"
+    python3 -m idf_component_manager create-remote-namespace \
+        --namespace="${NAMESPACE}"
+
+    check_if_exists
+
+    echo "Creating component ${NAME}"
+    python3 -m idf_component_manager create-remote-component \
```

Comment:
I would prefer not to create any components automatically. Right now it's possible to do it manually with the CLI later it will also be possible with the UI.

If a component with the given name doesn't exist - the job has to fail, but with a clear message.

Reply:
I agree we don't create namespace automatically. That's a bit too risky.

But for creating components, I think it's more intuitive to create the component automatically the first time we try to use this action?

Reply:
If we would agree on automatic components creation, then the change should be made on the server code, so it will create a component automatically on running "upload-component". Then no changes are required for the action. In this case, I would suggest closing this MR.

Reply:
Initially, we wanted to avoid automatically created components by the CLI, to avoid typos, however, it makes sense for the CI.